pallets/gas/src/lib.rs

```diff
@@ -208,30 +210,34 @@ pub mod pallet {
             });
 
             if delete_current_node {
-                ValueView::<T>::remove(key);
+                GasTree::<T>::remove(key);
             }
 
             res
         }
 
-        pub fn node_root_origin(node: &ValueNode) -> H256 {
+        /// Returns the AccountId (as Origin) of the value tree creator
+        pub(crate) fn node_root_origin(node: &ValueNode) -> H256 {
             match node.inner {
                 ValueType::External { id, .. } => id,
                 ValueType::SpecifiedLocal { parent, .. }
                 | ValueType::UnspecifiedLocal { parent } => {
-                    Self::node_root_origin(&Self::value_view(parent).expect("Parent should exist"))
+                    Self::node_root_origin(&Self::get_node(parent).expect("Parent should exist"))
```

Comment:
what about returning here Option<H256> if parent doesn't exist despite the fact that this should be in storage

Reply:
Our implementation guarantees a node can't be removed from the tree as long as it has at least one child. If the parent doesn't exist for a key while it has to, it means the storage is somehow corrupted. We don't have control over it as we only connect to the underlying storage through `sp_io::storage::Storage` runtime interface. Such database corruption can probably be the case on one validator (or a block producer) but not on all the validators in the network at once. Corrupted storage in any case means a node must be shut down and restarted. But the network should be able to recover.

Reply:
Here as well